### PR TITLE
feat: remove oldest grays

### PR DESCRIPTION
**Pull request by tanvibhakta**
## Description

This PR removes grays that were present in the very first color system. They have been replaced by grays in the current scale. 
Fixes #https://github.com/appsmithorg/appsmith/issues/17545

Components affected: 
Breadcrumbs
Button 
ButtonTab
Checkbox
Dropdown
EditableTextWrapper
EmojiReactions
FilePicker
GifPlayer
Menu
MenuDivider
Multiswitch
Radio
ScrollIndicator
SearchInput 
SegmentHeader
Switch
TableDropdown
TagInput
TextInput 
Toggle

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

These colors have been verified as okay by Vasanth. 

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/design-system/src/ListSegmentHeader/index.tsx [hunk 1]
```diff
@@ -34,9 +26,9 @@ export default function SegmentHeader(props: SegmentHeaderProps) {
       style={props.style}
     >
       {props.title}
-      {!props.hideStyledHr && (
-        <StyledHr data-testid={"t--styled-segment-header-hr"} />
```
Comment (albinAppsmith) [inline packages/design-system/src/ListSegmentHeader/index.tsx:38]: Why this is removed?

File: packages/design-system/src/Radio/index.tsx [hunk 2]
```diff
@@ -98,6 +101,7 @@ export const Radio = styled.label<{
 
   input:disabled ~ .checkbox:after {
     background-color: var(--ads-radio-disabled-background-color);
+    //border: var(--ads-radio-disabled-border-color);
```
Comment (albinAppsmith) [inline packages/design-system/src/Radio/index.tsx:104]: ?

**Review by albinAppsmith — CHANGES_REQUESTED** [review]

File: packages/design-system/src/ListSegmentHeader/index.tsx (hunk 1 above)
Comment (tanvibhakta) [reply packages/design-system/src/ListSegmentHeader/index.tsx:38]: There are no places where this is being used in the main repo right now. I was getting an error on local with this prop and removing this fixed it.

File: packages/design-system/src/Radio/index.tsx (hunk 2 above)
Comment (tanvibhakta) [reply packages/design-system/src/Radio/index.tsx:104]: Was trying something out. Will remove this, my bad.

File: packages/design-system/src/themes/default/colors.css [hunk 3]
```diff
@@ -35,6 +35,20 @@
   --ads-color-black-0 : #FFFFFF;
 
 
+  /* gray */
```
Comment (rohitagarwal88) [inline packages/design-system/src/themes/default/colors.css:38]: @tanvibhakta / @albinAppsmith : Shouldn't we will follow our new naming convention in greyscale? As we are changing variables across the ads & same variable will be used in main repo

**Review by rohitagarwal88 — CHANGES_REQUESTED** [review]

File: packages/design-system/src/ListSegmentHeader/index.tsx [hunk 4]
```diff
@@ -34,9 +26,9 @@ export default function SegmentHeader(props: SegmentHeaderProps) {
...
-      )}
+      {/*  TODO: Fix cypress here */}
```
Comment (albinAppsmith) [inline packages/design-system/src/ListSegmentHeader/index.tsx:29]: Why this commented?

File: packages/design-system/src/themes/default/colors.css (hunk 3 above)
Comment (albinAppsmith) [reply packages/design-system/src/themes/default/colors.css:38]: This is just for the colors. Variable naming convention is for global and component level variables.

**Review by albinAppsmith — CHANGES_REQUESTED** [review]

**Review by albinAppsmith — APPROVED** [review]

**Review by rohitagarwal88 — APPROVED** [review]

**Pull request merged**